context/FilterContext.jsx

```diff
@@ -0,0 +1,39 @@
+import {
+  createContext, useContext, useMemo, useState,
+} from 'react';
+
+const FilterContext = createContext();
+
+export const availabilityOptions = {
+  AVAILABLE: 'available',
+  UNAVAILABLE: 'unavailable',
+};
+
+export const useFilterContext = () => useContext(FilterContext);
+
+export function FilterProvider({ children }) {
+  const [availability, setAvailability] = useState(availabilityOptions.AVAILABLE);
+  const [project, setProject] = useState('all');
+  const [roles, setRoles] = useState([
+    'Developer',
+    'Director of Business Development',
+    'Frontend Designer',
+    'Project Manager',
+    'Technical Director',
+    'UI Designer',
+    'User Experience Strategist',
+    'Projects Director',
+  ]);
```

Comment:
This should work in such a way that if `roles` is an empty array (i.e. no checkboxes checked), then we should see everybody. Let's update this to be `useState([])`, then I'll leave a comment in the logic on how we could achieve our intended behavior.